src/utils/sdk.js

```diff
@@ -119,11 +119,21 @@ function expectedSDKVersion() {
   return version;
 }
 
+// Ensure that the user has a version of Xcode installed and usable.
 function ensureViableXCode() {
   const xcodeBuildExec = '/usr/bin/xcodebuild';
   if (fs.existsSync(xcodeBuildExec)) {
     const result = cp.spawnSync(xcodeBuildExec, ['-version']);
-    if (result.status === 0) return;
+    if (result.status === 0) {
+      const match = result.stdout
+        .toString()
+        .trim()
+        .match(/Xcode (\d+\.\d+)/);
+      if (match && !semver.satisfies(semver.coerce(match[1]), '>14')) {
+        fatal(`Xcode version ${version} is not supported, please upgrade to Xcode 15 or newer`);
+      }
+      return;
```

Comment:
```suggestion
      if (match) {
        if (!semver.satisfies(semver.coerce(match[1]), '>14')) {
          fatal(`Xcode version ${match[1]} is not supported, please upgrade to Xcode 15 or newer`);
        } else {
          return;
        }
      }
```

I led things slightly astray with my previous comment, but I think this cleans it up. We want to fall-through to the more detailed error message below if there's no `match`, if the version number either fails to satisfy the check we give the more specific error, else everything is as expected and return.